packages/meditrak-server/src/routes/importSurveyResponses/importSurveyResponses.js

```diff
@@ -38,183 +39,181 @@ export async function importSurveyResponses(req, res) {
-                const survey = await transactingModels.survey.findOne({ name: surveyName });
-                const entityCode = getInfoForColumn(sheet, columnIndex, 'Entity Code');
-                const entity = await transactingModels.entity.findOne({ code: entityCode });
-                const user = await transactingModels.user.findById(req.userId);
-                // 'Date of Data' is pulled from spreadsheet, 'Date of Survey' is current time
-                const surveyDate = getDateForColumn(sheet, columnIndex);
-                const importDate = moment();
-                const newSurveyResponse = await transactingModels.surveyResponse.create({
-                  survey_id: survey.id, // All survey responses within a sheet should be for the same survey
-                  assessor_name: `${user.first_name} ${user.last_name}`,
-                  user_id: user.id,
-                  entity_id: entity.id,
-                  start_time: importDate,
-                  end_time: importDate,
-                  data_time: stripTimezoneFromDate(surveyDate),
-                  timezone: timeZone,
-                });
-                newSurveyResponseIds[columnIndex] = newSurveyResponse.id;
-              } else {
-                // Validate that every header takes id form, i.e. is an existing or deleted response
-                await hasContent(columnHeader);
-                await takesIdForm(columnHeader);
-              }
-            } catch (error) {
-              throw new ImportValidationError(
-                `Invalid survey response id ${columnHeader} causing message: ${
-                  error.message
-                } at column ${columnIndex + 1} on tab ${tabName}`,
+      for (let columnIndex = 0; columnIndex <= maxColumnIndex; columnIndex++) {
+        const columnHeader = getColumnHeader(sheet, columnIndex);
+        if (!isInfoColumn(columnIndex)) {
+          // A column representing a survey response
+          try {
+            if (checkIsNewSurveyResponse(columnHeader)) {
+              const surveyResponseDetails = await constructNewSurveyResponseDetails(
+                models,
+                tabName,
+                sheet,
+                columnIndex,
+                config,
               );
+              updateBatcher.createSurveyResponse(tabName, columnHeader, surveyResponseDetails);
+            } else {
+              // Validate that every header takes id form, i.e. is an existing or deleted response
+              await hasContent(columnHeader);
+              await takesIdForm(columnHeader);
             }
+          } catch (error) {
+            throw new ImportValidationError(
+              `Invalid survey response id ${columnHeader} causing message: ${
+                error.message
+              } at column ${columnIndex + 1} on tab ${tabName}`,
+            );
           }
         }
+      }
+
+      const infoValidator = new ObjectValidator(constructInfoColumnValidators(models));
```

Comment:
There really aren't as many changes as this diff would have you believe, particularly in the section below this line - it just seems to have got the line matching a bit wrong. Might be easier to eyeball the two files side by side rather than trust the red and green